### PR TITLE
Copy subtype phpdoc on ClassPropertyAssignToConstructorPromotionRector (#8438)

### LINKED ISSUE #8438: Incorrect behavior of ClassPropertyAssignToConstructorPromotionRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

more precise php-doc - in this case `positive-int` - is lost while transformation

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/a37c4a63-360f-4d0a-b1f0-42eb2885841a

```php
<?php
namespace Fare\ArticleExport\Viewmodel;
/**
 * @immutable
 */
final class PreisStufeVM
{
    /** @var positive-int */
    public int $mengeab;

    public float $price;

    public function __construct(int $mengeab, float $price)
    {
        $this->mengeab = $mengeab;
        $this->price = $price;
    }
}
```

### Responsible rules

* `ClassPropertyAssignToConstructorPromotionRector`

### PR DESCRIPTION
**Pull request by pkvach**
Closes https://github.com/rectorphp/rector/issues/8438

### CODE REVIEW

File: rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/copy_integer_range_type_doc.php.inc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\Fixture;
+
+class RectorRun
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/copy_integer_range_type_doc.php.inc]:
```suggestion
class CopyIntegerRangeType
```

File: rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/copy_integer_range_type_doc.php.inc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\Fixture;
+
+class RectorRun
```
Comment (samsonasik) [inline rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/copy_integer_range_type_doc.php.inc]:
```suggestion
class CopyIntegerRangeType
```

**Comment by samsonasik** [comment]
Thank you @pkvach

**Comment by staabm** [comment]
Thanks 🙏

**Comment by pkvach** [comment]
Thank you, too

**Pull request merged**